runtime/wasm/src/host.rs

```diff
@@ -718,3 +719,16 @@ impl RuntimeHostTrait for RuntimeHost {
         .await
     }
 }
+
+impl PartialEq for RuntimeHost {
+    fn eq(&self, other: &Self) -> bool {
+        // mapping_request_sender, host_exports and host_metrics are operational structs not needed
+        // to define uniqueness; each runtime host should be for a unique data source.
+        self.data_source_name == other.data_source_name
+            && self.data_source_contract == other.data_source_contract
+            && self.data_source_contract_abi == other.data_source_contract_abi
+            && self.data_source_event_handlers == other.data_source_event_handlers
+            && self.data_source_call_handlers == other.data_source_call_handlers
+            && self.data_source_block_handlers == other.data_source_block_handlers
```

Comment:
Destructure `self` before comparing, to prevent this becoming incorrect when new fields are added.